src/custom_types.rs

```diff
@@ -0,0 +1,77 @@
+use lsp_data::Range;
```

Comment:
We ought to be able to derive both if we need to (I think we have feature gates for that, or maybe that is span and we could add them for data)